src/cogent3/core/new_alignment.py

```diff
@@ -706,33 +751,39 @@ def get_translation(
         Returns
         -------
         A new instance of self translated into protein
+
+        Notes
+        -----
+        Translating will break the relationship to an annotation_db if present.
         """
-        # todo: kath, add check for when include_stop==True, the seq.get_translation method should be
-        # returning sequences with a protein_with_stop moltype.
 
         if len(self.moltype.alphabet) != 4:
-            raise TypeError("Sequences must be a DNA/RNA")
+            raise new_alpha.AlphabetError("Sequences must be a DNA/RNA")
 
         translated = {}
-        if trim_stop and not include_stop:
-            seqs = self.trim_stop_codons(gc=gc, strict=not incomplete_ok)
-        else:
-            seqs = self
         # do the translation
-        for seqname in seqs.names:
-            seq = seqs.seqs[seqname]
+        for seqname in self.names:
+            seq = self.seqs[seqname]
             pep = seq.get_translation(
-                gc, incomplete_ok=True, include_stop=include_stop, trim_stop=trim_stop
+                gc,
+                incomplete_ok=incomplete_ok,
+                include_stop=include_stop,
+                trim_stop=trim_stop,
             )
-            translated[seqname] = pep
+            translated[seqname] = str(pep)
```

Comment:
add comment design, do we have to cast to string?

Reply:
Actually, we don't now that `Sequence`s can be cast to arrays